stm/stm.ml

```diff
@@ -2803,6 +2803,11 @@ let process_transaction ?(newtip=Stateid.fresh ()) ?(part_of_script=true)
       (* Meta *)
       | VtMeta, _ ->
         let id, w = Backtrack.undo_vernac_classifier expr in
+        (* Special case Backtrack, as it is never part of a script. See #6240 *)
+        let part_of_script = begin match Vernacprop.under_control expr with
```

Comment:
why patch here and not in the classifier itself?

Reply:
sorry, I forgot things are not like that anoymore

Reply:
So indeed we should think which "Meta" commands should be kept as part of the document vernacular itself, and which ones are really "Meta". AFAICT only Backtrack seems really META, right?

Reply:
I've never seen Backtrack written in a real document so I think it is meta (it is just concrete syntax for edit_at)

Reply:
Yeah I couldn't move it to the META level for 8.8 as #6970 arrived late.

Reply:
@gares what more of the commands treated by the undo_classifier should be moved to `toplevel`?